### PR TITLE
Fixed iOS 14 large title header being collapsed initially

### PR DESCRIPTION
**Pull request by aryanm5**
## Summary

The `largeTitle` option for scrollview/flatlist headers on iOS by [React Native Screens](https://github.com/software-mansion/react-native-screens) was initially rendering in its collapsed "small" state when the screen first opens. See: [#649](https://github.com/software-mansion/react-native-screens/issues/649) [#484](https://github.com/grahammendick/navigation/issues/484) [#645](https://github.com/software-mansion/react-native-screens/issues/645)

We must ensure that updateContentOffsetIfNeeded first runs after the ScrollView is moved to the window. So we prevent it running inside reactSetFrame if there’s no window yet and get it to run from didMoveToWindow instead. The problem is we need to stop the updateContentOffsetIfNeeded running in reactSetFrame because that's too early. You can see what that means by commenting out the !self.windowcheck in the fix and seeing it won't work anymore. 

This fix is only needed on iOS 14.

Expected behavior: The title should begin in large form and collapse only when the user scrolls down, to match the behavior of native iOS.

The following gif illustrates the problematic behavior. Notice how the "Items" header starts off in its collapsed form instead of large text.

problematic large title gif

This pull request fixes the issue by causing the title to start off in its fully expanded form.

## Changelog

[iOS] [Fixed] - Fix scrollview large title starting off collapsed on iOS 14.

## Test Plan

This is the behavior of the header after implementing the pull request. Notice that upon opening the second screen, the header starts off in its expanded form, which is expected behavior. This fix is only needed on iOS 14.

working large title gif

### CODE REVIEW

**Comment by trajano** [comment]
Would this require an update to the Expo SDK/Client to work since this appears to be a native code issue.

**Comment by trajano** [comment]
Would this require an update to the Expo SDK/Client to work since this appears to be a native code issue.

File: React/Views/ScrollView/RCTScrollContentView.m [hunk 1]
```diff
@@ -14,13 +14,20 @@
...
+}
+
 - (void)reactSetFrame:(CGRect)frame
 {
   [super reactSetFrame:frame];
 
   RCTScrollView *scrollView = (RCTScrollView *)self.superview.superview;
 
-  if (!scrollView) {
+  if (!scrollView || !self.window) {
```
Comment (philIip) [inline React/Views/ScrollView/RCTScrollContentView.m:30]: curious why this change is necessary - my understanding is that the bug happens due to a race of when we have a window after the initial call to this method happens.

if that's the case, then shouldn't we be good after we re-update the content size in didMoveToWindow? why do subsequent calls to reactSetFrame fail?

**Comment by fabOnReact** [comment]
Can you reproduce this issue only with react-native (without react-native-screens or other libraries)? Thanks

**Pull request closed without merging**